tip/environments.py

```diff
@@ -0,0 +1,85 @@
+import os
+import json
+
+from . import packages
+
+
+def locate(tip_dir: str, name: str) -> str:
+    """Find file of the environment called `name`."""
+    return os.path.join(_get_environments_dir(tip_dir), f"{name}.json")
+
+
+def get_environment_by_path(path: str) -> dict:
+    """Get package list of the environment at `path`."""
+    if not os.path.isfile(path):
+        raise FileNotFoundError(f"Environment file is not found at {path}")
+    with open(path, mode='r', encoding='utf8') as environment_file:
+        return json.load(environment_file)
```

Comment:
If function reads a file, it's not a getter and shouldn't be named with "get_" prefix. "get_" prefix suggests that it's O(1) operation.

Reply:
That's a great criteria for "get_" prefixed functions! Fixed.

Reply:
The criterion is that if you just return what is after "get_" then it's a getter, as it gets the value and returns it. If you do some computations to obtain X then it can be named `compute_X` or `calc_X`. If you load X from file then it should be named `load_X`.
That's why when an engineer used to such conventions sees a function with "get_" prefix they assume that it's lightweight and can be called millions of times without need of caching the value. If it's not true, it may cause performance issues at least.